core/src/test/java/org/elasticsearch/transport/netty/NettyTransportRequestLimitIT.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.transport.netty;
+
+import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.common.network.NetworkModule;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
+import org.elasticsearch.test.ESIntegTestCase.Scope;
+
+import java.net.InetAddress;
+import java.util.Locale;
+
+import static org.elasticsearch.common.settings.Settings.settingsBuilder;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+
+@ClusterScope(scope = Scope.SUITE, minNumDataNodes = 1, maxNumDataNodes = 5, numClientNodes = 0)
+public class NettyTransportRequestLimitIT extends ESIntegTestCase {
```

Comment:
I've moved this test to `CircuitBreakerServiceIT` but I am not sure whether we want it there. We have three possible client combinations that we could test: node client, transport client (local transport), transport client (netty transport). This test is irrelevant (and would also fail) with the node client so I had to prevent it (`@ClusterScope.transportClientRatio = 1`). 

If we had a possibility to randomize the transport (either local or netty) then it would make sense. The only possibilities I've found were explicit definition of a specific transport via system property or to prevent one of the two with `@ESIntegTestCase.SuppressLocalMode` or `@ESIntegTestCase.SuppressNetworkMode`. So I am not too sure whether it makes sense to move it to `CircuitBreakerServiceIT` in the end (or did I miss a possibility to randomize it except for adjusting settings manually?).